provider/server/machine_annotations_update_test.go

```diff
@@ -36,12 +36,11 @@ var _ = Describe("UpdateMachineAnnotations", func() {
 		Expect(createResp).NotTo(BeNil())
 
 		DeferCleanup(func(ctx SpecContext) {
-			Eventually(func(g Gomega) bool {
+			Eventually(func(g Gomega) {
 				_, err := machineClient.DeleteMachine(ctx, &iri.DeleteMachineRequest{MachineId: createResp.Machine.Metadata.Id})
-				g.Expect(err).To(SatisfyAny(
-					BeNil(),
-					MatchError(ContainSubstring("NotFound")),
-				))
+				g.Expect(err).Should(Succeed())
+			}).Should(Succeed())
```

Comment:
It can be simplified:
```suggestion
Eventually(func() error {
	_, err := machineClient.DeleteMachine(ctx, &iri.DeleteMachineRequest{MachineId: createResp.Machine.Metadata.Id})
	return err
}).Should(Succeed())
```
This eliminates the need for the additional Gomega argument as well. Please update other occurrences accordingly.

Reply:
what is benefits? it looks more as cosmetic change. both solution are valid for gomega and this solution is more similar another eventually blocks.

Reply:
The functionality remains the same, but this approach offers advantages in terms of:

Readability: It explicitly conveys that the DeleteMachine call is expected to eventually succeed, eliminating the need for an additional Expect statement.

Clarity: Using Eventually(func() error { ... }).Should(Succeed()) directly signifies the expectation that an operation will ultimately succeed, avoiding the need for explicit error handling within the function.

However, this approach is optional and may vary based on preference.

Reply:
i will implement it.